src/nvim/api/win_config.c

```diff
@@ -330,7 +341,88 @@ static bool parse_float_bufpos(Array bufpos, lpos_T *out)
   return true;
 }
 
-static void parse_border_style(Object style, FloatConfig *fconfig, Error *err)
+static void parse_border_title(Object title, Object title_pos, Object width, FloatConfig *fconfig,
+                               Error *err)
+{
+  if (!parse_title_pos(title_pos, fconfig, err)) {
+    return;
+  }
+
+  int win_width = (int)width.data.integer;
+
+  if (title.type == kObjectTypeString) {
+    fconfig->title_text = xstrdup(title.data.string.data);
+    int len = (int)strlen(fconfig->title_text);
+    if (len > win_width) {
+      api_set_error(err, kErrorTypeException, "title length must be less than win width");
+      return;
+    }
+
+    int hl_id = object_to_hl_id(title, "border title highlight", err);
+    fconfig->title_hi_id = hl_id;
+    fconfig->title = true;
+    return;
+  }
+
+  if (title.type != kObjectTypeArray) {
+    api_set_error(err, kErrorTypeValidation, "title must be string or array");
+    return;
+  }
+
+  if (title.type == kObjectTypeArray && title.data.array.size == 0) {
+    api_set_error(err, kErrorTypeValidation, "title cannot be an empty array");
+    return;
+  }
+
+  Array arr = title.data.array;
+  size_t size = arr.size;
+  for (size_t i = 0; i < size; i++) {
+    Object iytem = arr.items[i];
+    if (iytem.type != kObjectTypeArray || iytem.data.array.size == 0) {
+      api_set_error(err, kErrorTypeValidation, "not allow empty array in title");
+      return;
+    }
+
+    if (iytem.data.array.items[0].type != kObjectTypeString) {
+      api_set_error(err, kErrorTypeValidation, "missing title string in title array");
+      return;
+    }
+  }
+
+  fconfig->title_texts = parse_virt_text(title.data.array, err, &win_width);
+
+  fconfig->title = true;
+  return;
+}
+
+static bool parse_title_pos(Object title_pos, FloatConfig *fconfig, Error *err)
+{
+  if (!HAS_KEY(title_pos)) {
+    fconfig->title_pos = kBorderTitleLeft;
+    return true;
+  }
+
+  if (title_pos.type != kObjectTypeString) {
+    api_set_error(err, kErrorTypeValidation, "title_pos must be string");
+    return false;
+  }
+
+  char *pos = title_pos.data.string.data;
+
+  if (striequal(pos, "left")) {
```

Comment:
why case-insensitive? better to be restrictive and precise unless there is a reason to accept variants. Loose casing is extra complexity.

Reply:
good catch .  should be my mistake .